src/arch/ucx/machine.C

```diff
@@ -282,9 +398,25 @@ void LrtsInit(int *argc, char ***argv, int *numNodes, int *myNodeID)
         ucxCtx.eagerSize = UCX_MSG_PROBE_THRESH;
     }
 
+    // We wait for rxReqsBatch requests to complete, before reposting new ones
+    ucxCtx.rxReqsBatch = std::max(1, ucxCtx.numRxReqs / 4);
+    if (CmiGetArgInt(*argv, "+ucx_rx_reqs_batch", &ucxCtx.rxReqsBatch)) {
+        if ((ucxCtx.rxReqsBatch > ucxCtx.numRxReqs) || (ucxCtx.rxReqsBatch <= 0)) {
+            CmiPrintf("UCX: Invalid value of RX reqs batch: %d\n", ucxCtx.rxReqsBatch);
+            CmiAbort(__func__);
+        }
+    }
+
     UcxInitEps(*numNodes, *myNodeID);
 
-    UcxPrepostRxBuffers();
+    ucxCtx.rxReqs = (UcxRequest**)CmiAlloc(sizeof(UcxRequest*) * ucxCtx.numRxReqs);
+    CmiEnforce(ucxCtx.rxReqs);
+
+    // Post RX requests for direct protocol
```

Comment:
Since have a Direct API, the usage of 'direct' here is a little confusing. 

Can you change the comment to 'Post RX requests for large messages (which are over the eager threshold and use RDMA)'

Reply:
ok